projects/public-search/src/app/document-brief/document-brief.component.ts

```diff
@@ -16,17 +16,89 @@
  */
 
 import { Component, Input } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { RecordService as LocalRecordService } from '../record.service';
 
 @Component({
   selector: 'public-search-document-brief',
-  templateUrl: './document-brief.component.html'
+  templateUrl: './document-brief.component.html',
+  styles: [`
+.thumb-brief
+  img {
+    max-height: 90px;
+    width: 58px;
+  }
+
+@media (max-width: 960px) {
+  .thumb-brief
+    img {
+      max-width: 48px;
+    }
+  }
+
+pre {
+  white-space: pre-wrap;
+  max-height: 300px;
+  font-size: 0.7em;
+}
+`]
 })
 export class DocumentBriefComponent {
 
-  @Input()
-  record: any;
+  // TODO: adapt following line when issue #23 of ng-core is closed
+  private pathArray = window.location.pathname.split('/');
 
   @Input()
-  type: string;
+  set record(value) {
+    if (value !== undefined) {
+      this.Record = value;
+      this.coverUrl = `/static/images/icon_${value.metadata.type}.png`;
+      if (value.metadata.cover_art) {
+        this.coverUrl = value.metadata.cover_art;
+      } else if (value.metadata.identifiedBy) {
+        let isbn;
+        for (const identifier of value.metadata.identifiedBy) {
+          if (identifier.type === 'bf:Isbn') {
+            isbn = identifier.value;
+          }
+        }
+        if (isbn) {
+          this.getCover(isbn);
+        }
+      }
+    }
+  }
+
+  get record() {
+    return this.Record;
+  }
+
+  private Record: any;
+
+  coverUrl: string;
+
+  // TODO: adapt following line when issue #23 of ng-core is closed
+  public view = this.pathArray[1];
+
+  constructor(
+    private translate: TranslateService,
+    private localRecordService: LocalRecordService
+  ) {  }
+
+  authorName(author) {
+    const nameIndex = `name_${this.translate.currentLang}`;
+    let nameLng = author[nameIndex];
+    if (!nameLng) {
+      nameLng = author.name;
+    }
+    return nameLng;
+  }
 
+  getCover(isbn) {
```

Comment:
could you add headers for each methods, containing at least the description of what do the method. And try to type arguments (string, boolean, ...) when it's possible.